pkg/steps/template.go

```diff
@@ -591,6 +598,38 @@ func waitForPodCompletionOrTimeout(ctx context.Context, podClient coreclientset.
 	}
 }
 
+func removeFinalizer(ctx context.Context, podClient coreclientset.PodInterface, pod *coreapi.Pod, finalizer string) {
+	if ctx.Err() != nil {
+		// give ourselves a bit of a grace period to remove the finalizer
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(context.Background(), 5*time.Minute)
+		defer cancel()
+	}
+
+	nextFinalizers := make([]string, 0, len(pod.ObjectMeta.Finalizers))
+	for _, entry := range pod.ObjectMeta.Finalizers {
```

Comment:
Can be simpliefied as `sets.NewString(pod.FInalizers).Delete(finalizer).List()`

Reply:
>    sets.NewString(pod.FInalizers).Delete(finalizer).List()

Does that preserve order?  Because I want to delete our entry without shuffling the remaining entries, if any have been added out of band.

Reply:
A very explicit mention about finalizers is that order must not matter (because if it did, everyone would want to be last)

Reply:
Ref: https://github.com/kubernetes/kubernetes/pull/83110/files#diff-0bc759fffb6a495573c6b70a9afbf741R666

Reply:
I'm not saying it has semantic meaning, I'm just saying I would like to avoid unneccessary churn.  Maybe someone is diffing YAML or some such to track changes, etc.

Reply:
Ppl are not supposed to rely on implementation details like this and this is the only finalizer that could be there

Reply:
It's possible that admission hooks or other out-of-band actors add their own finalizers.  And yeah, while ideally folks keep track of which properties are semantically ordered and which are unordered, it's still cheap to avoid churning even the unordered properties.

Reply:
There is no ideally. Finalizer order must not matter. Its explicitly in the upstream docs. Please change this to simplify the code instead of writing complicated code to account for behavior that is explicitly documented as wrong.